ui/src/app/components/copy-modal.tsx

```diff
@@ -21,54 +23,111 @@ const ResourceTypeHomeTabs = new Map()
   .set(ResourceType.DATASET, 'data');
 
 export interface Props {
+  cdrVersionListResponse: CdrVersionListResponse;
   fromWorkspaceNamespace: string;
-  fromWorkspaceName: string;
+  fromWorkspaceFCName: string;
   fromResourceName: string;
+  fromCdrVersionId: string;
   onClose: Function;
   onCopy: Function;
   resourceType: ResourceType;
   saveFunction: (CopyRequest) => Promise<FileDetail | ConceptSet>;
 }
 
+interface WorkspaceOptions {
+  label: string;
+  options: Array<{label: string, value: Workspace}>;
+}
+
 interface State {
-  writeableWorkspaces: Array<Workspace>;
+  workspaceOptions: Array<WorkspaceOptions>;
   destination: Workspace;
   newName: string;
   requestState: RequestState;
-  errorMsg: string;
+  copyErrorMsg: string;
   loading: boolean;
 }
 
-const boldStyle = {
-  fontWeight: 600
-};
+const styles = reactStyles({
+  bold: {
+    fontWeight: 600
+  },
+});
 
-class CopyModal extends React.Component<Props, State> {
+class CopyModalComponent extends React.Component<Props, State> {
   constructor(props: Props) {
     super(props);
 
     this.state = {
-      writeableWorkspaces: [],
+      workspaceOptions: [],
       newName: props.fromResourceName,
       destination: null,
       requestState: RequestState.UNSENT,
-      errorMsg: '',
-      loading: true
+      copyErrorMsg: '',
+      loading: true,
     };
   }
 
+  cdrName(cdrVersionId: string): string {
+    const {cdrVersionListResponse} = this.props;
+    return cdrVersionListResponse.items.find(version => version.cdrVersionId === cdrVersionId).name;
+  }
+
+  groupWorkspacesByCdrVersion(workspaces: Workspace[]): Array<WorkspaceOptions> {
+    const {fromCdrVersionId} = this.props;
+    const workspacesByCdr = fp.groupBy(w => w.cdrVersionId, workspaces);
+    const cdrVersions = Array.from(new Set(workspaces.map(w => w.cdrVersionId)));
+
+    // list the "from" CDR version first.
+    const fromCdrVersionFirst = (cdrv1: string, cdrv2: string) => {
+      if (cdrv1 === fromCdrVersionId && cdrv2 !== fromCdrVersionId) {
+        return -1;
+      } else if (cdrv1 !== fromCdrVersionId && cdrv2 === fromCdrVersionId) {
+        return 1;
+      } else {
+        // TODO: a meaningful ordering, possibly as part of RW-5563
```

Comment:
I'd recommend the list CDRs provide an index or date  you can use for sorting so you don't get into this lexical sorting business. We can't keep them from coming up with new version string formats, so this could get hairy fast.